barretenberg/cpp/src/barretenberg/bb/main.cpp

```diff
@@ -516,27 +516,43 @@ void vk_as_fields(const std::string& vk_path, const std::string& output_path)
  *
  * @param bytecode_path Path to the file containing the serialised bytecode
  * @param calldata_path Path to the file containing the serialised calldata (could be empty)
- * @param crs_path Path to the file containing the CRS (ignition is suitable for now)
+ * @param public_inputs_path Path to the file containing the serialised avm public inputs
+ * @param hints_path Path to the file containing the serialised avm circuit hints
  * @param output_path Path (directory) to write the output proof and verification keys
  */
 void avm_prove(const std::filesystem::path& bytecode_path,
                const std::filesystem::path& calldata_path,
+               const std::filesystem::path& public_inputs_path,
+               const std::filesystem::path& hints_path,
                const std::filesystem::path& output_path)
 {
     // Get Bytecode
-    std::vector<uint8_t> const avm_bytecode =
+    std::vector<uint8_t> const bytecode =
         bytecode_path.extension() == ".gz" ? gunzip(bytecode_path) : read_file(bytecode_path);
-    std::vector<uint8_t> call_data_bytes{};
-    if (std::filesystem::exists(calldata_path)) {
-        call_data_bytes = read_file(calldata_path);
+
+    std::vector<uint8_t> calldata_bytes;
+    try {
+        calldata_bytes = read_file(calldata_path);
+    } catch (std::runtime_error const& err) {
+        vinfo("No calldata was provided for avm proving.... Might be fine!");
+    }
+    std::vector<fr> const calldata = many_from_buffer<fr>(calldata_bytes);
+
+    std::vector<uint8_t> const public_inputs_bytes = read_file(public_inputs_path);
```

Comment:
If you don't need the bytes you can just do `std::vector<fr> const public_inputs_vec = many_from_buffer<fr>(read_file(public_inputs_path));`